framework/source/class/qx/util/ColorUtil.js

```diff
@@ -164,83 +166,63 @@ qx.Bootstrap.define("qx.util.ColorUtil",
 
 
     /**
-     * Try to convert an incoming string to an RGB array.
-     * Supports themed, named and system colors, but also RGB strings,
-     * hex3 and hex6 values.
+     * Try to convert an incoming string to an RGBA array.
+     * Supports themed, named and system colors, but also RGBA strings,
+     * hex[3468] values.
      *
      * @param str {String} any string
-     * @return {Array} returns an array of red, green, blue on a successful transformation
+     * @return {Array} returns an array of red, green, blue and optional alpha on a successful transformation
      * @throws {Error} if the string could not be parsed
      */
     stringToRgb : function(str)
     {
       if (this.supportsThemes() && this.isThemedColor(str)) {
         str = qx.theme.manager.Color.getInstance().resolveDynamic(str);
       }
-
-      if (this.isNamedColor(str))
-      {
-        return this.NAMED[str].concat();
-      }
-      else if (this.isSystemColor(str))
-      {
-        throw new Error("Could not convert system colors to RGB: " + str);
-      }
-      else if (this.isRgbaString(str)) {
-        return this.__rgbaStringToRgb(str);
-      }
-      else if (this.isRgbString(str))
-      {
-        return this.__rgbStringToRgb();
-      }
-      else if (this.isHex3String(str))
-      {
-        return this.__hex3StringToRgb();
-      }
-      else if (this.isHex6String(str))
-      {
-        return this.__hex6StringToRgb();
-      }
-
-      throw new Error("Could not parse color: " + str);
+      return this.cssStringToRgb(str);
     },
 
-
     /**
-     * Try to convert an incoming string to an RGB array.
-     * Support named colors, RGB strings, hex3 and hex6 values.
+     * Try to convert an incoming string to an RGB array with optional alpha.
+     * Support named colors, RGB strings, RGBA strings, hex[3468] values.
      *
      * @param str {String} any string
      * @return {Array} returns an array of red, green, blue on a successful transformation
      * @throws {Error} if the string could not be parsed
      */
     cssStringToRgb : function(str)
     {
+      var color;
       if (this.isNamedColor(str))
       {
-        return this.NAMED[str];
+        color = this.NAMED[str].concat();
       }
       else if (this.isSystemColor(str))
       {
-        throw new Error("Could not convert system colors to RGB: " + str);
+          throw new Error("Could not convert system colors to RGB: " + str);
```

Comment:
Why is it an error if it's a system color?

Reply:
I have no idea ... that has been there before